src/VirtoCommerce.CatalogModule.Web/Controllers/Api/CatalogModuleListEntryController.cs

```diff
@@ -262,6 +230,50 @@ public async Task<ActionResult> Move([FromBody]ListEntriesMoveRequest moveReques
             return NoContent();
         }
 
+        /// <summary>
+        /// Bulk delete by the search criteria.
+        /// </summary>
+        /// <param name="criteria"></param>
+        [HttpPost]
+        [Route("delete")]
+        [ProducesResponseType(typeof(void),StatusCodes.Status204NoContent)]
+        public async Task<ActionResult> Delete([FromBody] CatalogListEntrySearchCriteria criteria)
+        {
+            const int deleteBatchSize = 20;
+            var authorizationResult = await _authorizationService.AuthorizeAsync(User, criteria, new CatalogAuthorizationRequirement(ModuleConstants.Security.Permissions.Delete));
+            if (!authorizationResult.Succeeded)
+            {
+                return Unauthorized();
+            }
+
+            var idsToDelete = criteria.ObjectIds?.ToList() ?? new List<string>();
+
+            if (idsToDelete.IsNullOrEmpty())
+            {
+                var listEntries = await InnerListItemsSearchAsync(criteria);
+                idsToDelete = listEntries.ListEntries
+                    .Where(x => x.Type.EqualsInvariant(KnownDocumentTypes.Product) || x.Type.EqualsInvariant(KnownDocumentTypes.Category))
```

Comment:
use `ProductListEntry.TypeName` and `CategoryListEntry.TypeName` instead of `KnownDocumentTypes`

Reply:
used